### ISSUE TITLE
508-defect-2 [SCREENREADER]: On this page navigation should use <nav> instead of <section>

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#18628]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a user navigating a page using a screen reader, I want to be able to find the "on this page" navigation by regions.

## Details

Since the "on this page" navigation is a navigational region, we should use `<nav>` instead of `<section>` . This will allow screen reader users who scan through pages by regions better identify and jump to the "on this page" navigation. It will also make for better semantic markup 👍 

## Acceptance Criteria

- [ ] `<nav>` is used instead of `<section>` 
- [ ] Screen reader users can distinguish the "on this page" `<nav>` from other `<nav>` s on the page

## Steps to Recreate

1. Enter `https://wave.webaim.org/report#/https://www.va.gov/coronavirus-veteran-frequently-asked-questions-esp/` in browser
2. Click on the "Structure" tab
3. Confirm the "on this page" navigation is not listed as a navigation

## Proposed Solution

We can fix this by changing the `<section>` to a `<nav>` and by making use of `aria-labelledby` to differentiate it from other `<nav>`s on the page.

```html
<section id="table-of-contents">
  <h2 class="vads-u-margin-bottom--2 vads-u-font-size--lg" id="on-this-page">
    On this page
  </h2>
</section>
```

```html
<nav id="table-of-contents" aria-labelledby="on-this-page">
  <h2 class="vads-u-margin-bottom--2 vads-u-font-size--lg" id="on-this-page">
    On this page
  </h2>
</nav>
````

## WCAG or Vendor Guidance

* [W3C- Labeling Regions](https://www.w3.org/WAI/tutorials/page-structure/labels/)

## Screenshots or Trace Logs

<img width="1567" alt="Screen Shot 2021-01-15 at 11 43 38 AM" src="https://user-images.githubusercontent.com/14154792/104754380-26217500-5727-11eb-9615-54c9278212cd.png">

<img width="827" alt="Screen Shot 2021-01-15 at 11 51 53 AM" src="https://user-images.githubusercontent.com/14154792/104755387-5c132900-5728-11eb-9e49-68d9e17bc624.png">

**Comment by DanielleThierryUSDSVA** [comment]
@SKasimow @youngfreezyVA This is another sitewide Drupal component issue.

**Comment by SKasimow** [comment]
@youngfreezyVA check on whether there is a site-wide solution to address under dev.

**Comment by youngfreezyVA** [comment]
i can fix for this specific page... these are all the other places in the codebase that we are using the "on this page". it is possible for me to fix there as well... please let me know if i should proceed @SKasimow 
![Uploading Screen Shot 2021-01-28 at 8.14.30 AM.png…]()

**Comment by joshkimux** [comment]
@1Copenut , flagging this for your awareness! I remember we talked about whether or not the accessibility guidance for this component was set in place yet-- but it looks like it hasn't at a glance.

**Comment by youngfreezyVA** [comment]
yeah... i think there's a larger issue at play here. why so many divs with the same id and sometimes a class? not sure we can even consider this a component, it's more like scattered markup throughout the codebase.

**Comment by youngfreezyVA** [comment]
@ncksllvn @kelsonic tagging you guys in case i'm missing something... here are the changes i was going to make in a branch for just the FAQ page:
https://github.com/department-of-veterans-affairs/vets-website/compare/master...18628-18527-On-This-Page

Nick, I am not sure if I can/should also make this change in the same file you pointed me to earlier:
https://github.com/department-of-veterans-affairs/vets-website/blob/0fe779514bcb1c09a769df0f6267e1ff32287299/src/site/stages/build/plugins/modify-dom/add-id-to-subheadings.js#L94

**Comment by ncksllvn** [comment]
Your change looks like it's in the right place. It will be a site-wide change, but if the `<nav/>` element makes more sense on the FAQ page it probably make more sense for all the other pages too
